### PR TITLE
Solve JSONDecodeError

### PR DESCRIPTION
**Pull request by HuZixia**
Solve JSONDecodeError, about issue #749 

To avoid JSONDecodeError:

Remove comments in output json str, after json value content, maybe start with #, maybe start with //, particularly, it is not inside the string value

Addtionly, if you do not want JSONDecodeError to occur, you can add 'Delete comments in json' after FORMAT_CONSTRAINT in action_node.py

**Features**

The json content returned by a LLM may contain comments, maybe start with #, maybe start with //, it's random. This can lead to subsequent json parsing errors that affect the overall code execution. 
The Error has happened whether I use gpt-4-1106-preview or GLM-4. 
These code changes are intended to fix this problem.

**Feature Docs**

**Influence**

These code changes are intended to fix JSONDecodeError.

**Result**

See the issue #749 in detail

The json comments maybe start with #
image

The json comments maybe start with //
image

So we need to fix the code repair_llm_raw_output.py.

Add the second way to fix bug about raised JSONDecodeError:
Add prompt "Delete comments in json" of FORMAT_CONSTRAINT in action_node.py, as follows:

raised JSONDecodeError as follows:
image

add "Delete comments in json", solve the problem:
image

### CODE REVIEW

File: metagpt/utils/repair_llm_raw_output.py [hunk 1]
```diff
@@ -198,6 +206,21 @@ def repair_invalid_json(output: str, error: str) -> str:
...
+                comment_index = -1
+                for match in re.finditer(r"(\".*?\"|\'.*?\')|(#|//)", line):
+                    if match.group(1):  # if the string value
+                        continue
+                    if match.group(2):  # if comments
+                        comment_index = match.start(2)
+                        break
+                # if comments, then delete them
+                if comment_index != -1:
+                    arr[i] = arr[i][:comment_index].rstrip()
```
Comment (geekan) [inline metagpt/utils/repair_llm_raw_output.py:223]: The new two pieces of logic are the same, would it be better to merge them?

Comment (HuZixia) [reply metagpt/utils/repair_llm_raw_output.py]: Okay, it's a good idea, I merged the code and resubmitted it. Please have a look, thanks.

File: PR/# remove comments in output json str, after js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+
+git commit -m "To avoid JSONDecodeError: " -m "Remove comments in output json str, after json value content, maybe start with #, maybe start with //, particularly, it is not inside the string value" -m "Addtionly, if you do not want JSONDecodeError to occur, you can add 'Delete comments in json' after FORMAT_CONSTRAINT in action_node.py"
```
Comment (geekan) [inline PR/# remove comments in output json str, after js:2]: These three files appear to be the wrong submissions

Comment (HuZixia) [reply PR/# remove comments in output json str, after js:2]: Yeah, sorry, do you want me to delete it? I need to see how to do.

Comment (geekan) [reply PR/# remove comments in output json str, after js:2]: @HuZixia yes. delete PR dir

File: metagpt/utils/repair_llm_raw_output.py [hunk 3]
```diff
@@ -105,6 +105,23 @@ def judge_potential_json(routput: str, left_key: str) -> Union[str, None]:
     return output
 
 
+def remove_comments_from_line(line):
```
Comment (better629) [inline metagpt/utils/repair_llm_raw_output.py:108]: add unittest case for `//` like `https://github.com/geekan/MetaGPT/blob/main/tests/metagpt/utils/test_repair_llm_raw_output.py#L131-L141`

File: PR/# remove comments in output json str, after js (hunk 2 above)
Comment (HuZixia) [reply PR/# remove comments in output json str, after js]: Done～

File: metagpt/actions/action_node.py [hunk 4]
```diff
@@ -23,7 +23,10 @@
 TAG = "CONTENT"
 
 LANGUAGE_CONSTRAINT = "Language: Please use the same language as Human INPUT."
-FORMAT_CONSTRAINT = f"Format: output wrapped inside [{TAG}][/{TAG}] like format example, nothing else."
+FORMAT_CONSTRAINT = (f"Format: output wrapped inside [{TAG}][/{TAG}] like format example, nothing else. "
+                     f"Delete comments in json")
```
[the remark this answers is not in the document]
Comment (HuZixia) [reply metagpt/actions/action_node.py:27]: Well, I agree with you.

File: metagpt/utils/repair_llm_raw_output.py (hunk 3 above)
Comment (HuZixia) [reply metagpt/utils/repair_llm_raw_output.py:108]: OK

Comment (HuZixia) [reply metagpt/utils/repair_llm_raw_output.py]: Modify code based on feedback of action_node.py and repair_llm_raw_output.py, add code in test_repair_llm_raw_output.py. Please have a look, thanks.

**Comment by better629** [comment]
LGTM

**Review by geekan — APPROVED** [review]
LGTM

**Pull request merged**